### PR TITLE
Security Enhancement - secures DB dumps

### PR DESCRIPTION
**Pull request by dg4prez**
Prevents access levels lower than oper (20) from accessing database dump functionality.

### CODE REVIEW

File: traffic_ops/app/lib/UI/GenDbDump.pm [hunk 1]
```diff
@@ -32,6 +33,10 @@ sub dbdump {
 		$self->internal_server_error( { Error => "Error dumping database" } );	
 		return;
 	}
+	if ( !&is_oper($self) ) {
```
Comment (mitchell852) [inline traffic_ops/app/lib/UI/GenDbDump.pm]: why don't you do this permissions check before you do the actual dump?

Comment (dg4prez) [inline traffic_ops/app/lib/UI/GenDbDump.pm]: Very good point.  Moved.

File: traffic_ops/app/lib/UI/GenDbDump.pm [hunk 2]
```diff
@@ -17,11 +17,17 @@ package UI::GenDbDump;
...
 use Mojo::Base 'Mojolicious::Controller';
 use Data::Dumper;
+use UI::Utils;
 
 sub dbdump {
 	my $self = shift;
 	my $filename = $self->param('filename');
 
+	if ( !&is_oper($self) ) {
+		$self->internal_server_error( { Error => "Insufficient permissions for DB Dump. Admin access is required." } );
```
Comment (mitchell852) [inline traffic_ops/app/lib/UI/GenDbDump.pm]: your error message is not quite right :)

Admin OR ops role is required...

Comment (mitchell852) [inline traffic_ops/app/lib/UI/GenDbDump.pm]: unless you want to change that to is_admin()

Comment (dg4prez) [inline traffic_ops/app/lib/UI/GenDbDump.pm]: Fixed.

**Pull request closed without merging**